src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/Utilities.cs

```diff
@@ -104,4 +106,25 @@ private static async Task RetryAsync(Func<Task<Exception?>> executor, ITestOutpu
             exception = await executor();
         }
     }
+
+    public static void LogWarningMessage(this ITestOutputHelper outputHelper, string message)
+    {
+        string prefix = "##vso[task.logissue type=warning;]";
+
+        outputHelper.WriteLine($"{Environment.NewLine}{prefix}{message}.{Environment.NewLine}");
+        outputHelper.WriteLine("##vso[task.complete result=SucceededWithIssues;]");
+    }
+
+    public static void LogAndThrowIfNullOrEmpty(this ITestOutputHelper outputHelper, string? variable, string variableName)
+    {
+        if (string.IsNullOrEmpty(variable) || string.IsNullOrWhiteSpace(variable))
```

Comment:
Isn't the `IsNullOrEmpty` check redundant because IsNullOrWhiteSpace checks to empty as well?

```suggestion
        if (string.IsNullOrWhiteSpace(variable))
```